app/code/community/Ebizmarts/MailChimp/Helper/Data.php

```diff
@@ -1378,40 +1378,52 @@ protected function getLastPromoCodeSent($scopeId, $scope)
      */
     public function createMergeFields($scopeId, $scope)
     {
+        $success = 1;
         $listId = $this->getGeneralList($scopeId, $scope);
         $maps = $this->unserialize($this->getMapFields($scopeId, $scope));
         $customFieldTypes = $this->unserialize($this->getCustomMergeFieldsSerialized($scopeId, $scope));
 
-        try {
-            $api = $this->getApi($scopeId, $scope);
-            $mailchimpFields = array();
-
+        if (count($maps) > 30 ) {
+            $success = 2;
+        } else {
             try {
-                $mailchimpFields = $api->getLists()->getMergeFields()->getAll($listId, null, null, 50);
-            } catch (MailChimp_Error $e) {
-                $this->logError($e->getFriendlyMessage());
-            }
+                $api = $this->getApi($scopeId, $scope);
+                $mailchimpFields = array();
 
-            if (!empty($mailchimpFields)) {
-                foreach ($maps as $map) {
-                    $customAtt = $map['magento'];
-                    $chimpTag = $map['mailchimp'];
-                    $alreadyExists = false;
+                try {
+                    $mailchimpFields = $api->getLists()->getMergeFields()->getAll(
+                        $listId,
+                        null,
+                        null,
+                        50
+                    );
+                } catch (MailChimp_Error $e) {
+                    $this->logError($e->getFriendlyMessage());
+                }
 
-                    foreach ($mailchimpFields['merge_fields'] as $mailchimpField) {
-                        if ($mailchimpField['tag'] == $chimpTag || strtoupper($chimpTag) == 'EMAIL') {
-                            $alreadyExists = true;
+                if (!empty($mailchimpFields)) {
+                    foreach ($maps as $map) {
+                        $customAtt = $map['magento'];
+                        $chimpTag = $map['mailchimp'];
+                        $alreadyExists = false;
+
+                        foreach ($mailchimpFields['merge_fields'] as $mailchimpField) {
+                            if ($mailchimpField['tag'] == $chimpTag || strtoupper($chimpTag) == 'EMAIL') {
+                                $alreadyExists = true;
+                            }
                         }
-                    }
 
-                    if (!$alreadyExists) {
-                        $this->_createCustomFieldTypes($customFieldTypes, $api, $customAtt, $listId, $chimpTag);
+                        if (!$alreadyExists) {
+                            $this->_createCustomFieldTypes($customFieldTypes, $api, $customAtt, $listId, $chimpTag);
+                        }
                     }
                 }
+            } catch (Ebizmarts_MailChimp_Helper_Data_ApiKeyException $e) {
```

Comment:
Make sure to add Mailchimp_Error catch below this one in case **_createCustomFieldTypes** throws it.
After all the catch in the controller will be removed if the success value is given in this function.